### ISSUE TITLE
overflow ellipsis applies when it should not [5.3.0-alpha6]

### ISSUE THREAD

**Issue by christophe-g** [vaadin/vaadin-grid#1523]
overflow ellipsis renders in cases where I think it should not, as illustrated below (content of the cell actually does not overflow - happens on lumo and material - chrome - related: #1304).  

![image](https://user-images.githubusercontent.com/1168053/50249449-62fece80-03de-11e9-85c7-1219c5e925dd.png)

the column in question:
```
<vaadin-grid-column width="60px" flex-grow="0">
        <template>
          <paper-icon-button  icon="more-vert"  alt="menu" on-tap="activateContextMenu"></paper-icon-button>
        </template>
      </vaadin-grid-column>
```
As a workaround, I'll use `cell-class-name-generator` to apply a class to this column and override the ellipsis styling. 

As a remark, `cell-class-name-generator` offers a lot of flexibility. However, in most of the cases you would like to apply the same class all column cells, ideally in a **declarative** way (in the markup). Smth like: 
```html
<vaadin-grid-column width="60px" flex-grow="0" class="i-can-use-this-class-without-a-function">
        <template>
          <paper-icon-button  icon="more-vert"></paper-icon-button>
        </template>
</vaadin-grid-column>
```

And then
```css
[part~="cell"].i-can-use-this-class-without-a-function ::slotted(vaadin-grid-cell-content) {
       text-overflow: initial;
}
```
would be nice ; )

**Comment by masbaehr** [comment]
Happens for me too but with the selector checkboxes :o 

![image](https://user-images.githubusercontent.com/38694044/51334171-8ff6dd80-1a7f-11e9-99c2-6318348f71ae.png)

**Comment by web-padawan** [comment]
Another reproduction is here: https://github.com/vaadin/vaadin-grid/issues/1561#issue-408840654

**Comment by silvan-lincan** [comment]
Continuing the investigation from #1561, by using the Lumo compact theme/preset, the checkboxes are now rendered Ok. Maybe this helps further.

**Comment by web-padawan** [comment]
@silvan-lincan actually the problem is what Grid Flow has its own selection column implementation and that one has not been updated to align with the changes in the web component 🤦‍♂️

**Comment by web-padawan** [comment]
@christophe-g regarding your issue: agree that it seems confusing, but that's because of `vaadin-grid-cell-content` horizontal padding, which you can indeed tweak by using cell class generators.

**Comment by christophe-g** [comment]
@web-padawan - thanks !
> which you can indeed tweak by using cell class generators.

That's what I have been doing, but it does not solve the root cause of the issue, which I tentatively remember as : ellipsis appearing in cases where cell content does not overflow.

**Comment by web-padawan** [comment]
The issue described in https://github.com/vaadin/vaadin-grid/issues/1523#issuecomment-455242769 is fixed by vaadin/vaadin-grid-flow#520

Regarding the `paper-icon-button`, the only thing that I can recommend would be custom theme with overriding the following styles: https://github.com/vaadin/vaadin-grid/blob/master/theme/lumo/vaadin-grid-styles.html#L45-L48
